graphql/resolvers.go

```diff
@@ -60,6 +63,49 @@ func (r *queryResolver) Task(ctx context.Context, taskID string) (*model.APITask
 	return &apiTask, nil
 }
 
+func (r *queryResolver) Projects(ctx context.Context) ([]*GroupedProjects, error) {
+	allProjs, err := model.FindAllTrackedProjectRefs()
+	if err != nil {
+		return nil, errors.Wrap(err, "error retrieving projects")
+	}
+
+	groupsMap := make(map[string][]*restModel.UIProjectFields)
+
+	for _, p := range allProjs {
+		groupName := s.Join([]string{p.Owner, p.Repo}, "/")
+
+		uiProj := restModel.UIProjectFields{
+			DisplayName: p.DisplayName,
+			Identifier:  p.Identifier,
+			Repo:        p.Repo,
+			Owner:       p.Owner,
+		}
+
+		if projs, ok := groupsMap[groupName]; ok {
+			groupsMap[groupName] = append(projs, &uiProj)
+		} else {
+			groupsMap[groupName] = []*restModel.UIProjectFields{&uiProj}
+		}
+	}
+
+	groupsArr := []*GroupedProjects{}
+
+	for groupName, groupedProjects := range groupsMap {
+		name := groupName
+		gp := GroupedProjects{
+			Name:     &name,
+			Projects: groupedProjects,
+		}
+		groupsArr = append(groupsArr, &gp)
+	}
+
+	sort.SliceStable(groupsArr, func(i, j int) bool {
+		return *groupsArr[i].Name < *groupsArr[j].Name
```

Comment:
is it possible for an element in groupsArr to be nil?

Reply:
there should never be a nil element